.github/workflows/run_tests.yml

```diff
@@ -30,7 +30,27 @@ jobs:
         with:
           command: "run run_tests.m"
       - uses: codecov/codecov-action@v2
-        if: ${{ matrix.os == 'ubuntu-latest' && matrix.matlab_version == 'latest' }}
+        if: ${{ always() && matrix.os == 'ubuntu-latest' && matrix.matlab_version == 'latest' }}
         with:
             files: coverage*.xml
-            token: ${{ secrets.CODECOV_TOKEN }}
\ No newline at end of file
+            token: ${{ secrets.CODECOV_TOKEN }}
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: Unit test results ${{ matrix.os }}
+          path: junit_report*.xml
+
+  publish-test-results:
```

Comment:
Note the `run_tests.m` functions as expected giving a `junit_report*.xml` that has the same filename for subsequent runs on my local computer (and is therefor overwritten each time). I guess this is to avoid having to do a cleanup of xml files before running the tests?

Reply:
If a test fails it'll edit the post in this PR - I forced a fail here to see if it would work
![image](https://user-images.githubusercontent.com/18187336/172194923-b295a1f1-f91d-4043-9f8a-f5f1fcb11e6d.png)